fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ConstraintExceptionMapper.java

```diff
@@ -51,15 +51,29 @@
     public static Link buildConstraintLink(final ConstraintViolationException e,
```

Comment:
Why not just change this to be: `public static Link buildConstrainLink(final Throwable e, [...]);`?

Reply:
I think the reason for that (though I didn't write that part) is because only constraint violations will have the actual RDF file backing the URL to be created here.

Reply:
Right, but you refactored it to add a method that accepts a string class name. I could keep the code the same and just change the method signature without having to add a new method.

Reply:
Ok, so the reason for the string is only because I couldn't (easily) de-duplicate the Exceptions because they all have different messages and so remained distinct. So I map all the exceptions to their class name and get the distinct set of those. Then I needed to build the constraint link using the class name which is why I added the new String method. I was not opening this up to any exception out there.

Reply:
Gotcha. Now that the exceptions are defined as a set of `ConstraintViolationException`. You could change this signature to:

```java
public static Link buildConstraintLink(final Class<? extends ConstraintViolationException> e, final ServletContext context, final UriInfo uriInfo);
```

and then collected like:

```java
     final Link[] constraintLinks = e.getExceptionTypes().stream().map(Exception::getClass)
                .distinct().map(c -> buildConstraintLink(c, context, uriInfo))
                .toArray(Link[]::new);
```